### PR TITLE
Spellchecking compiler literals

### PR DESCRIPTION
**Pull request by nyurik**
This PR cleans up the spelling mistakes in the compiler literals.

### CODE REVIEW

File: compiler/rustc_lint/src/builtin.rs [hunk 1]
```diff
@@ -332,7 +332,7 @@ impl UnsafeCode {
         self.report_unsafe(cx, span, |lint| {
             lint.build(msg)
                 .note(
-                    "the linker's behavior with multiple libraries exporting duplicate symbol \
+                    "the linker behavior with multiple libraries exporting duplicate symbol \
```
Comment (compiler-errors) [inline compiler/rustc_lint/src/builtin.rs:335]: Is this a typo?

Comment (nyurik) [reply compiler/rustc_lint/src/builtin.rs:335]: it feels incorrect grammatically, but i'm not an expert. Spellchecker didn't like it either.  Not sure

Comment (Dylan-DPC) [reply compiler/rustc_lint/src/builtin.rs:335]: it still makes sense grammatically though in this case

**Comment by compiler-errors** [comment]
Re: discussion on #95497, let's close this one per the spellcheck PR policy decided by the compiler team, since it didn't go through an intensive review yet.

**Pull request closed without merging**